File: dogshelter_mobile/lib/features/donations/domain/kategorija_donacije.dart
class KategorijaDonacije {
  KategorijaDonacije({required this.kategorijaDonacijeId, required this.naziv, required this.ikonaKljuc});

  final int kategorijaDonacijeId;
  final String naziv;
  final String ikonaKljuc;

  bool get isOstalo => naziv == 'Ostalo';

  factory KategorijaDonacije.fromJson(Map<String, dynamic> json) {
    return KategorijaDonacije(
      kategorijaDonacijeId: json['kategorijaDonacijeId'] as int,
      naziv: json['naziv'] as String,
      ikonaKljuc: json['ikonaKljuc'] as String,
    );
  }
}

File: DogShelter/DogShelter.Model/KategorijaDonacije.cs
namespace DogShelter.Model;

public class KategorijaDonacije
{
    public int KategorijaDonacijeId { get; set; }
    public string Naziv { get; set; } = null!;
    public string IkonaKljuc { get; set; } = null!;
    public int? PodrazumijevanaJedinicaMjereId { get; set; }

    // Empty means unrestricted (e.g. "Ostalo") - the client should then allow any unit.
    public List<JedinicaMjere> DozvoljeneJedinice { get; set; } = [];
}
